tools/src/launcher/Main.hs

```diff
@@ -515,11 +519,47 @@ spawnNode (path, args, mbLogPath) = do
             logInfo "Node has started"
             return (ph, asc)
 
-runWallet :: (FilePath, [Text]) -> M ExitCode
-runWallet (path, args) = do
+runWallet :: Bool -> (FilePath, [Text], Maybe FilePath) -> M ExitCode
+runWallet shouldLog (path, args, mLogPath) = do
     logNotice "Starting the wallet"
-    view _1 <$>
-        liftIO (readProcessWithExitCode path (map toString args) mempty)
+    phvar <- newEmptyMVar
+    liftIO $ case mLogPath of
+        Just lp -> do
+            createDirectoryIfMissing True (directory lp)
+            (_, logHandle) <- (lp,) <$> openFile lp AppendMode
+            liftIO $ IO.hSetBuffering logHandle IO.LineBuffering
+            let cr = createProc (Process.UseHandle logHandle) path args
+            system' phvar cr mempty EWallet
+        Nothing ->
+           let cr = if shouldLog then
+                        Process.Inherit
+                    else Process.NoStream
+           in liftIO $ system' phvar (createProc cr path args) mempty EWallet
+
+createProc :: Process.StdStream -> FilePath -> [Text] -> Process.CreateProcess
+createProc stdStream path args =
+    (Process.proc path (map toString args))
+            { Process.std_in = Process.CreatePipe
+            , Process.std_out = stdStream
+            , Process.std_err = stdStream
+            }
+
+asyncLog :: Handle -> Handle -> ProcessHandle -> Executable -> IO ExitCode
+asyncLog stdO stdE pH nt = do
+    withAsync (forever $ IO.hGetLine stdO >>= customLogger LInfo nt) $ \_ ->
+        withAsync (forever $ IO.hGetLine stdE >>= customLogger LError nt) $ \_ -> do
```

Comment:
`Data.Text.IO` from `text` package also has `hGetLine` function which should be more efficient. In general we tend to avoid `String` unless there is no way to avoid it.